account_financial_report/report/aged_partner_balance.py

```diff
@@ -165,7 +166,7 @@ def _get_move_lines_data(
         move_lines = [
             move_line
             for move_line in move_lines
-            if move_line["date"] <= date_at_object
+            if (move_line[date_by] or move_line["date"]) <= date_at_object
```

Comment:
```suggestion
            if ((move_line[date_by] == "invoice_date" and move_line["invoice_date"]) or move_line["date"]) <= date_at_object
```
same in open_items.py

Reply:
There are some cases where invoice_date field can be empty because that I use this comparison to avoid comparison error with boolean value

Reply:
But this comparison is wrong, as `move_line[date_by]` it's always set. Also my suggestion is correct, because if invoice_date is empty, then `or move_line["date"]` will take its place.

Reply:
But move_line[date_by] data depends of value of date_by, and date_by can be for this case "date" or "invoice_date" for data in first case I'm sure always retrieve move_line["date"] in second case move_line["invoice_date"] can be a False or None value, and or helps to get always move_line["date"], maybe I'm missing something?